homeassistant/components/history_stats/data.py

```diff
@@ -131,6 +131,18 @@ async def async_update(
             await self._async_history_from_db(
                 current_period_start_timestamp, current_period_end_timestamp
             )
+            if event and (new_state := event.data["new_state"]) is not None:
+                if (
+                    current_period_start_timestamp
+                    <= floored_timestamp(new_state.last_changed)
+                    <= current_period_end_timestamp
+                ):
+                    self._history_current_period.append(
+                        HistoryState(
+                            new_state.state, new_state.last_changed.timestamp()
```

Comment:
```suggestion
                            new_state.state, new_state.last_changed_timestamp
```

Reply:
That is likely to be a cache hit

Reply:
I was not clear there was a cache here, or what exactly is being cached. This is just a copy from line 122 of the same thing. Is line 122 incorrect as well, or is this different for some reason?

Reply:
Line 122 should be updated as well.

This code is old and was written before `last_changed_timestamp` existed. Its not wrong, just slower than needed